go/libraries/doltcore/sqle/lookup/range.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lookup
+
+import (
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/typed/noms"
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+// Range represents the contiguous set of values that a lookup operation covers.
+type Range struct {
+	LowerBound Cut
+	UpperBound Cut
+}
+
+var (
+	// alwaysContinueRangeCheck will allow the range to continue until the end is reached.
+	alwaysContinueRangeCheck noms.InRangeCheck = func(tuple types.Tuple) (bool, error) {
+		return true, nil
+	}
+	// neverContinueRangeCheck will immediately end.
+	neverContinueRangeCheck noms.InRangeCheck = func(tuple types.Tuple) (bool, error) {
+		return false, nil
+	}
+)
+
+// OpenRange returns a range representing {l < x < u}.
+func OpenRange(lower, upper types.Tuple) Range {
+	return Range{
+		Above{key: lower},
```

Comment:
Request OpenLower() and ClosedLower(), OpenUpper(), etc. aliases for these constructors. Trying to sort these out as is makes my brain hurt.

Reply:
I could probably just rename these altogether.